src/content/api/parser.md

```diff
@@ -37,279 +38,477 @@ as such:
 
 `SyncBailHook`
 
-Evaluate the type of an identifier.
+Triggered when evaluating an expression consisting in a `typeof` of a free variable
 
-Parameters: `expression`
+- Hook Parameters: `identifier`
+
+- Callback Parameters: `expression`
+
+```js
+parser.hooks.evaluateTypeof.for('myIdentifier').tap('MyPlugin', expression => {
+  /* ... */
+  return expressionResult;
+});
+```
+
+```js
+// This will trigger the evaluateTypeof hook
```

Comment:
Suggest we move the comments out of code blocks and change it to:
```
This will trigger the evaluateTypeof hook

...js example

This wont trigger:

...js example
```